core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -1539,8 +1500,14 @@ sealed trait ZIO[-R, +E, +A] extends Serializable { self =>
               else ZIO.halt(loserCause && cause)
           }
       }
+
     val g = (b: B, a: A) => f(a, b)
-    (self raceWith that)(coordinate(f, true), coordinate(g, false))
+    (self raceWith that)(coordinate(f, true), coordinate(g, false)).fork.flatMap { f =>
+      f.await.flatMap { exit =>
```

Comment:
This looks like new functionality—isn't `raceWith` supposed to do the inheritance?

Reply:
This was from a previous implementation I did which was not on the run loop, I forgot to remove it.

Reply:
Without this fix, a test in `FiberRefSpec` fails:

```scala
testM("nothing gets inherited with a failure in zipPar") {
            for {
              fiberRef <- FiberRef.make(initial)
              success  = fiberRef.set(update)
              failure1 = fiberRef.set(update1) *> ZIO.fail(":-(")
              failure2 = fiberRef.set(update2) *> ZIO.fail(":-O")
              _        <- success.zipPar(failure1.zipPar(failure2)).orElse(ZIO.unit)
              value    <- fiberRef.get
            } yield assert(value, equalTo(initial))
          }
```
I wasn't sure breaking existing behavior was the way to go. This test assumes that if one of fibers fail, we don't carry the fiberRef the successful either, even if it won the race. What do you think should be done here?

Reply:
@YuvalItzchakov zip and race are different operations. Failure in zip cancels the computation and prevents ref inheritance – there is no "winner" in a failed zip.

Reply:
@neko-kai As long as zip uses race underneath, it's going to need to apply these additional semantics. That's why I added added this change to compensate.